### PR TITLE
pass mercurylookup boolean to checkUpkeep

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

**Comment by RyanRHall** [comment]
@FelixFan1992 what is the purpose of this boolean? It seems like we're expecting the caller to tell the registry wrapper if it should handle a mercury lookup or not, but my understanding was that only mercury server will know if an upkeep is allowed access or not.

File: pkg/types/types_test.go [hunk 1]
```diff
@@ -75,6 +78,7 @@ func TestDecode(t *testing.T) {
 			GasOverheadPerUpkeep: 300_000,
 			MaxUpkeepBatchSize:   1,
 			ReportBlockLag:       0,
+			MercuryLookup:        false,
```
Comment (infiloop2) [inline pkg/types/types_test.go:81]: not sure if already covered, but could you ensure that the missing value case is tested? (false being the default)

Comment (FelixFan1992) [reply pkg/types/types_test.go:81]: i believe it's already covered in this test where all fields are missing and the expected default is false.
another similar test is the one above where only MercuryLookup is missing and expected value is false.

**Comment by FelixFan1992** [comment]
Tested with chaincli, set config with MercuryLookup true and log the bool value

**Review by EasterTheBunny — APPROVED** [review]

**Pull request merged**